### ISSUE TITLE
build hangs without feedback when a context does not have a verified fingerprint for ssh

### ISSUE THREAD

**Issue by akutruff** [docker/buildx#1799]
### Contributing guidelines

- [X] I've read the [contributing guidelines](https://github.com/docker/buildx/blob/master/.github/CONTRIBUTING.md) and wholeheartedly agree

### I've found a bug and checked that ...

- [X] ... the documentation does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem

### Description

Relevant to this:
https://github.com/docker/for-win/issues/13488

After updating to 4.19 docker build hangs with no feedback forever. docker buildx ls also hangs forever. docker ps and docker run hello-world run perfectly fine.

I was using a *local* default context inside of windows, but had several other remote contexts defined behind ssh.  After 4.19, windows build commands started hanging.   Only after I happen to kill docker desktop did the error actually get spit out.  Was dumb luck.  This was not happening before.

### Expected behaviour

no hang - errors telling me that there is no ssh fingerprint

### Actual behaviour

it hangs

### Buildx version

10.04

### Docker info

```text
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.10.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.17.3
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.19
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v0.1.0-beta.4
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scan: Docker Scan (Docker Inc.)
    Version:  v0.26.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scan.exe
  scout: Command line tool for Docker Scout (Docker Inc.)
    Version:  v0.10.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 7
  Running: 3
  Paused: 0
  Stopped: 4
 Images: 7
 Server Version: 23.0.5
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 2806fc1057397dbaeefbea0e4e17bddfbd388f38
 runc version: v1.1.5-0-gf19387a
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 5.15.90.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 15.57GiB
 Name: docker-desktop
 ID: a6f6db06-57f6-45da-aaa9-d6f88717288f
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Registry: https://index.docker.io/v1/
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```
```


### Builders list

```text
redacted
```

### Configuration

redacted

**Comment by nc-jeni** [comment]
Any updates on this?

I have the same problem:
* Docker Desktop 4.19.0 (106363)
* Windows 11

**Comment by crazy-max** [comment]
@akutruff Do you have the output of `docker buildx ls` (please redact sensitive info)? Is it hanging to for this command too?

**Comment by akutruff** [comment]
@crazy-max Images taken from a WSL ubuntu prompt.  Windows command prompt is presently hanging again.  

![image](https://github.com/docker/buildx/assets/874049/f9b8bb82-1624-4a15-bd73-4094b371bbf2)

![image](https://github.com/docker/buildx/assets/874049/9fd9288b-5712-41e4-89ec-66376af256ce)

**Comment by akutruff** [comment]
@crazy-max reproducing if one of the context hosts is in unreachable as well.

**Comment by agalazis** [comment]
It also hangs when files are not present at all on linux host. Took me some time until I noticed that an old image created was actually copying some files and I forgot to provide the context.

**Comment by akutruff** [comment]
@crazy-max This issue is actually still happening even when all the hosts are reachable and all fingerprints have been accepted.  I confirmed that `docker ps` works fine for every context I have but when running `docker buildx ls`  it hangs.

**Comment by crazy-max** [comment]
The hang came from docker/cli's ssh connhelper blocking on an interactive host-key prompt or an unreachable host for `ssh://` contexts; buildx runs non-interactively, so host keys must be accepted beforehand (`ssh <endpoint>` once). Since v0.32.0 (f4cc8638c) the builder status timeout is configurable with `--timeout` on `ls`, `inspect`, `du`, `prune`, `rm` and `create`, and the reporter's residual hang went away after a Docker update.

Closing as stale. If this still happens on a current buildx and Docker Engine, please open a new issue with `docker buildx version`, `docker context ls` and `--debug` output.
